extensions/catalog-transfer-extension/src/main/java/sender/RegisterResourceRequestSender.java

```diff
@@ -46,42 +53,159 @@ public RegisterResourceRequestSender(ObjectMapper objectMapper) {
+        var dataSubcategory = getAssetDataSubcategory(registerResourceMessage);
+        var dataModel = getAssetDataModel(registerResourceMessage);
+        var geoReferenceMethod = getAssetGeoReferenceMethod(registerResourceMessage);
 
         var resource = new ResourceBuilder(registerResourceMessage.affectedResourceUri())
-                ._title_(new TypedLiteral(assetTitle, "en"))
-                ._description_(new TypedLiteral(assetDescription, "en"))
-                ._language_(Language.EN)
+                ._title_(new TypedLiteral(assetTitle, language.name()))
+                ._description_(new TypedLiteral(assetDescription, language.name()))
+                ._language_(language)
+                ._version_(version)
+                ._keyword_(keywords)
+                ._publisher_(URI.create(publisher))
+                ._standardLicense_(URI.create(standardLicense))
+                ._representation_(new RepresentationBuilder()
+                        ._language_(language)
+                        ._mediaType_(new IANAMediaTypeBuilder()._filenameExtension_(mediaType).build())
+                        .build())
+                ._resourceEndpoint_(List.of(new ConnectorEndpointBuilder()
+                        ._endpointDocumentation_(URI.create(endpointDocumentation))
+                        ._accessURL_(registerResourceMessage.affectedResourceUri())
+                        .build()))
                 .build();
-        return objectMapper.writeValueAsString(resource);
+
+        ObjectNode json = (ObjectNode) objectMapper.readTree(objectMapper.writeValueAsString(resource));
+        json.set("http://w3id.org/mds#transportMode", getTransportModeJson(transportMode));
+        json.set("http://w3id.org/mds#dataCategory", getDataCategoryJson(dataCategory));
+        json.set("http://w3id.org/mds#dataSubcategory", getDataSubcategoryJson(dataSubcategory));
+        json.set("http://w3id.org/mds#dataModel", getDataModelJson(dataModel));
+        json.set("http://w3id.org/mds#geoReferenceMethod", getGeoReferenceMethodJson(geoReferenceMethod));
+        return objectMapper.writeValueAsString(json);
+    }
+
+    private ObjectNode getGeoReferenceMethodJson(String geoReferenceMethod) {
+        return getPropertyJson(geoReferenceMethod);
+    }
+
+    private ObjectNode getDataModelJson(String dataModel) {
+        return getPropertyJson(dataModel);
+    }
+
+    private ObjectNode getDataSubcategoryJson(String dataSubcategory) {
+        return getPropertyJson(dataSubcategory);
+    }
+
+    private ObjectNode getTransportModeJson(String transportMode) {
+        return getPropertyJson(transportMode);
+    }
+
+    private ObjectNode getDataCategoryJson(String dataCategory) {
+        return getPropertyJson(dataCategory);
+    }
+
+    private ObjectNode getPropertyJson(String property) {
+        ObjectNode transportModeJson = objectMapper.createObjectNode();
```

Comment:
Maybe _var_ and why is the variable called `transportModeJson`?

Reply:
Good point, I forgot to change it when refactoring the method. It's called "json" now.